Added c2cx config code

briancaine opened the pull request:
Addresses #224

briancaine committed:
Extended config.Config to specify c2cx client settings

briancaine committed:
Added exchangeClient to passthrough.go

gz-c commented on src/config/config.go:
```diff
@@ -195,6 +197,27 @@ func (c SkyExchanger) validateWallet() []error {
+type ExchangeClient struct {
+	Key                      string `mapstructure:"key"`
+	Secret                   string `mapstructure:"secret"`
+	OrdersRefreshInterval    uint   `mapstructure:"orders_refresh_interval"`
```
use `time.Duration`

briancaine replied:
https://github.com/skycoin/teller/pull/227/commits/309aa620cded18fe78bdf6aadc908bb0c1cea1e4

Done.

gz-c commented on src/exchange/passthrough.go:
```diff
@@ -38,6 +40,11 @@ func NewPassthrough(log logrus.FieldLogger, cfg config.SkyExchanger, store Store
 		return nil, err
 	}
 
+	orderbookDatabase, err := db.NewOrderbookTracker()
```
Future concern: Is the orderbook tracker necessary? Can't we query the orderbook each time?

briancaine replied:
Yeah, I think you mentioned this before. It might not be necessary, or at the very least, we can add a construction function (instead of creating the `c2cx.Client` manually) and hide the db creation there.

If you make an issue on exchange-api and assign it to me, I can do the refactoring.

gz-c commented:
I see the build is failing due to 

```
$ go get -t ./...
# github.com/skycoin/teller/src/exchange
src/exchange/passthrough.go:56:17: cannot use *c2cx.Client literal (type *c2cx.Client) as type "github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".Client in field value:
	*c2cx.Client does not implement "github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".Client (wrong type for Buy method)
		have Buy(string, "github.com/skycoin/exchange-api/vendor/github.com/shopspring/decimal".Decimal, "github.com/skycoin/exchange-api/vendor/github.com/shopspring/decimal".Decimal) (int, error)
		want Buy(string, "github.com/skycoin/teller/vendor/github.com/shopspring/decimal".Decimal, "github.com/skycoin/teller/vendor/github.com/shopspring/decimal".Decimal) (int, error)
src/exchange/passthrough.go:61:10: cannot use "github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".NewTracker() (type "github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".Orders) as type "github.com/skycoin/exchange-api/exchange".Orders in field value:
	"github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".Orders does not implement "github.com/skycoin/exchange-api/exchange".Orders (wrong type for GetOrderInfo method)
		have GetOrderInfo(int) ("github.com/skycoin/teller/vendor/github.com/skycoin/exchange-api/exchange".Order, error)
		want GetOrderInfo(int) ("github.com/skycoin/exchange-api/exchange".Order, error)
```

Not sure what the best solution is for this.

If `vendor/github.com/skycoin/exchange-api/vendor` exists, we can try removing it and see if that fixes it.

Otherwise we may need to remove the `vendor/` from `exchange-api`.  If we do this, we keep the `Gopkg.toml` file (and maybe the `Gopkg.lock`?) but don't commit the `vendor/` folder. Then add README instructions for fetching dependencies.  Have to check the `dep` docs for the best practices when not committing `vendor/`.  `vendor/` only matters when using the `exchange-api` `cmd/` commands, which isn't important (nobody's using them) so we can require them to sync `vendor/` themselves (even add it as part of the `Makefile`)

gz-c commented:
Looks like it already prunes vendor, maybe its the way we run it in travis?

gz-c commented:
#232 is merged which removes `go get -t ./...`; it was unnecessary given vendoring (its purpose is to install all dependencies).  rebase/merge develop and try again

briancaine committed:
Merge branch 'develop' into implementing-passthrough

briancaine committed:
Updated dependencies

briancaine committed:
Replaced uint with time.Duration

Addressing this comment: https://github.com/skycoin/teller/pull/227#discussion_r173619521

briancaine commented:
Merged and addressed the comment about `time.Duration`.
